aspnet/6-pubsub/lib/BookDetailLookup.cs

```diff
@@ -223,15 +225,61 @@ public static void ProcessBook(IBookStore bookStore, long bookId)
             var response = WebRequest.Create(query).GetResponse();
             var reader = new StreamReader(response.GetResponseStream());
             var json = reader.ReadToEnd();
-            dynamic results = JObject.Parse(json);
-            dynamic info = results.items[0].volumeInfo;
-            book.Title = info.title;
-            book.PublishedDate = info.publishedDate;
-            book.Author = string.Join(", ", info.authors);
-            book.Description = info.description;
+            ParseBook(json, ref book);
             // TODO: update image.
             bookStore.Update(book);
         }
         // [END processbook]
+
+        /// <summary>
+        /// Parse a date time.  Return null if it can't be parsed.  A single number will be 
+        /// interpreted as a year, and the date returned will be YEAR-01-01.
+        /// </summary>
+        /// <param name="dateString">A string representation of the date.</param>
+        /// <returns>A datetime or null.</returns>
+        public static DateTime? ParseDate(string dateString)

```

Comment:
> What other formats have you seen / why would it return anything else?
> 
> All of the sample responses that I've seen in the Books API documentation use "YYYY" or "YYY-MM-DD"

That's all we've seen so far, but the API makes no guarantee that's all we'll ever see.  Failing to parse a date string is OK.  Failing to update the book just because we failed to parse a date string is what was happening, and what I'm trying to avoid.
 
> Volume resource representation
> Using the API - Volume responses
> If we want the application to support any type of date in the volumeInfo.publishedDate field then it should not be represented by our sample as a DateTime?.
> 
> Most of the Getting Started tutorial Bookshelf applications represent this data as a string. See the relevant lines of code:
> 
> python
> node
> go
> Ruby is the only other Bookshelf that parses publishedDate as a date object although, maybe that should be changed.
> 
> Book.PublicationDate being a DateTime? is not a required feature of the Bookshelf application, so I might vote that all Bookshelf applications should store publishedDate as a string because it greatly simplifies the code.

Perhaps in a later revision?  That's a bigger change than this PR.
 
> It's very unfortunate to have Date parsing logic in a sample that's supposed to demonstrate how to update a Book model from the Books API in the simplest possible way. Let's just make it a string - it results in the simplest code!

This code is not displayed to users on cloud.google.com.  I think most users will see the call to UpdateBookFromJson and never actually look to see the implementation.  It's not relevant to what they're trying to do.